src/coordinates/uniform_cartesian.hpp

```diff
@@ -166,6 +166,40 @@ class UniformCartesian {
     }
   }
 
+  template <int dir, TopologicalElement el>
+  KOKKOS_FORCEINLINE_FUNCTION Real X(const int idx) const {
+    using TE = TopologicalElement;
+    bool constexpr X1EDGE = el == TE::F1 || el == TE::E2 || el == TE::E3 || el == TE::NN;
+    bool constexpr X2EDGE = el == TE::F2 || el == TE::E3 || el == TE::E1 || el == TE::NN;
+    bool constexpr X3EDGE = el == TE::F3 || el == TE::E1 || el == TE::E2 || el == TE::NN;
+    if constexpr (dir == X1DIR && X1EDGE) {
+      return xmin_[dir - 1] + idx * dx_[dir - 1]; // idx - 1/2
+    } else if constexpr (dir == X2DIR && X2EDGE) {
+      return xmin_[dir - 1] + idx * dx_[dir - 1]; // idx - 1/2
+    } else if constexpr (dir == X3DIR && X3EDGE) {
+      return xmin_[dir - 1] + idx * dx_[dir - 1]; // idx - 1/2
+    } else {
+      return xmin_[dir - 1] + (idx + 0.5) * dx_[dir - 1]; // idx
+    }
+    return 0;
```

Comment:
is this default return statement necessary with the `else` clause?

Reply:
Logically it isn't necessary, but I wasn't sure if it would trigger a warning with some compilers so I put it in there.

Reply:
yeah, it appears removing this causes warnings in CUDA 11.4 at least